doc/deployment/deploy_custom.md

```diff
@@ -11,11 +22,360 @@ This document  details the various options of the `pachctl deploy custom ...` co
+
+
+#### Output formats
+
+- `--dry-run`: Create a manifest and send it to standard output, but do not deploy to Kubernetes.
+- `-o` or `--output`: An output format. You can choose from JSON (default) or YAML.
+
+#### Logging
+- `log-level`: The `pachd` verbosity level, from most verbose to least. You can set this parameter to `debug`, `info`, or `error`.
+- `-v` or `--verbose`: Controls the verbosity of the `pachctl` invocation.
+
+#### Complete example invocation
+Since we're deploying to a code infrastructure that works with YAML files,
+you need to add the flags for that.
+Limit your configuration to local roles only by adding the `--local-roles` flag.
+Because you need to deploy your cluster to a code infrastructure that works with YAML files,
+the output will be saved to a file that the code infrastructure can later use.
+The final deployment command includes the following flags:
+```
+pachctl deploy custom --persistent-disk aws --object-store s3 \
+    foobar 10 \
+    pachyderm-bucket  'OBSIJRBE0PP2NO4QOA27' 'tfteSlswRu7BJ86wekitnifILbZam1KYY3TG' 'minio:9000' \
+    --dynamic-etcd-nodes 10
+    --local-roles --output yaml  --dry-run > custom_deploy.yaml
+```
+
+For more information about the contents of the `custom_deploy.yaml` file,
+see [Anatomy of a Pachyderm deployment manifest](#anatomy-of-a-pachyderm-deployment-manifest).
+
+## Anatomy of a Pachyderm deployment manifest
+
+By default, when you run the `pachctl deploy ...` command with the `--dry-run` flag,
+you are generating a JSON-encoded Kubernetes manifest in one stream to standard output. 
+That manifest consists of smaller manifests,
+each of which corresponds to a particular aspect of a Pachyderm deployment.
+
+Pachyderm deploys the following sets of application components:
+- `pachd`, the main Pachyderm pod
+- `etcd`, the administrative datastore for `pachd`
+- `dash`, the web-based enterprise ui for Pachyderm Enterprise Edition
+
+In general, there are two categories of manifests in the file,
+roles-and-permissions-related and application-related. 
+
+
+## Roles and permissions manifests
+
+This section walks you through the roles and permissions manifests.
+
+### ServiceAccount
+
+Typically at the top of the file, a roles and permissions manifest has the `kind` key set to `ServiceAccount`. 
+[ServiceAccounts](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) are a way Kubernetes can assign namespace-specific privileges to applications in a lightweight way.
+The Pachyderm's service account is called  `pachyderm`.
+
+### Role or ClusterRole
+
+Depending on whether you used the `--local-roles` flag or not, the next manifest `kind` is `Role` or `ClusterRole`.
+
+### RoleBinding or ClusterRoleBinding
+
+This manifest binds the `Role` or `ClusterRole` to the `ServiceAccount` created above.
+
+## Application-related
```

Comment:
all the headings under this *Application-related* can be put in a table. We also need an intro text before the table. Something like:
```
This table lists all the application-related sections that Pachyderm configures in the deployment manifest.
```